internal/testrunner/runners/system/servicedeployer/custom_agent.go

```diff
@@ -0,0 +1,131 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package servicedeployer
+
+import (
+	_ "embed"
+	"os"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+
+	"github.com/elastic/elastic-package/internal/install"
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+//go:embed custom-agent-base-config.yml
+var customAgentYml string
+
+// CustomAgentDeployer knows how to deploy a custom elastic-agent defined via
+// a Docker Compose file.
+type CustomAgentDeployer struct {
+	cfg string
+	cd  *DockerComposeServiceDeployer
+}
+
+type deployedCustomAgent struct {
+	cfg string
+	*dockerComposeDeployedService
+}
+
+var _ DeployedService = &deployedCustomAgent{}
+
+// NewCustomAgentDeployer returns a new instance of a deployedCustomAgent.
+func NewCustomAgentDeployer(cfgPath string, ymlPaths []string, sv ServiceVariant) (*CustomAgentDeployer, error) {
+	path, err := createCustomAgentYaml(cfgPath)
+	if err != nil {
+		return nil, err
+	}
+
+	appConfig, err := install.Configuration()
+	if err != nil {
+		return nil, errors.Wrap(err, "can't read application configuration")
+	}
+
+	cd, err := newDockerComposeServiceDeployer(
+		append(ymlPaths, path),
+		appConfig.StackImageRefs(install.DefaultStackVersion).AsEnv(),
```

Comment:
Does it mean that it won't be possible to use other stack versions?

Reply:
Other stacks could be used by adding the `image` option in the custom-agent.yml file or using service variants.

Reply:
Out of curiosity: does it mean that it's possible to boot up Elastic stack 8.1 and the agent 8.3? I'm wondering if we need a helper to select the proper stack. See [kind deployer](https://github.com/elastic/elastic-package/blob/main/internal/testrunner/runners/system/servicedeployer/kubernetes.go#L151)